crates/ide-db/src/rename.rs

```diff
@@ -335,7 +334,14 @@ pub fn source_edit_from_references(
             _ => false,
         };
         if !has_emitted_edit && !edited_ranges.contains(&range.start()) {
-            edit.replace(range, new_name.to_string());
+            let new_name = match name {
+                ast::NameLike::Lifetime(_) => {
+                    format!("'{}", new_name.trim_start_matches("'"))
+                }
+                _ => new_name.into(),
+            };
+
+            edit.replace(range, new_name);
```

Comment:
```suggestion
            let (range, new_name) = match name {
                ast::NameLike::Lifetime(_) => (
                    TextRange::new(range.start() + syntax::TextSize::from(1), range.end()),
                    new_name.strip_prefix('\'').unwrap_or(new_name).to_owned(),
                ),
                _ => (range, new_name.to_owned()),
            };

            edit.replace(range, new_name);
```
We can omit the `'` here from the edited range